Automattic/wp-calypso
# Live Preview: apply selected style variation to the preview

Related to:

- https://github.com/Automattic/wp-calypso/issues/82914

## Proposed Changes

This PR supports applying the selected style variation to the Block Theme Previews. This works by updating the previewed theme's global styles CPT (post with type=`wp_global_styles`) (creating it first if it does not exist already) with the selected style variation. Note that even without this feature, a global styles CPT will be always created when a user live-preview a theme.

## Testing Instructions

1. Start with a site with Free plan. It's best to use a brand new site, so that it only has global styles CPT of the currently active theme.
2. Preview a theme WITHOUT style variations; verify that it still works.
3. Preview a theme with style variations, but without selecting any non-default variation; verify that it still works.
4. Preview the above theme but with a preselected non-default variation; verify that the variation is applied when the Site Editor loads.
5. Verify that the preselected variation is also selected in the `Styles` menu from the left sidebar:
    <img width="604" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/8f0564f6-a474-445a-8c25-265482b2ce39">
6. Preview the same theme but now with a default variation; verify that it still works (the default variation is applied).
7. Preview the same theme with another different non-default variation. Activate the theme.
8. Activate another theme.
9. Activate the previous theme WITHOUT selecting any variation; verify that you see the previous variation.
10. Upgrade the site to be an Atomic site.
11. Repeat and verify the steps above.


## Pre-merge Checklist
